seahorn/stubs/trusty_messaging_api_override.c

```diff
@@ -0,0 +1,98 @@
+/** Verify Trusty
+
+    Implementation of TrustyOS Messaging API
+    https://source.android.com/security/trusty/trusty-ref#messaging_api
+
+ */
+
+#include "handle_table.h"
+#include "nondet.h"
+#include "sea_mem_helper.h"
+#include "seahorn/seahorn.h"
+
+#include <stdint.h>
+#include <stdlib.h>
+#include <sys/types.h>
+
+// trusty reference for definitions only
+#include <trusty_ipc.h> // -> ipc structs
+#include <uapi/err.h>   // NO_ERROR definition
+
+static uint32_t cur_msg_id;
+static bool cur_msg_retired = true;
+
+static bool msg_retired(uint32_t msg_id) {
+  return msg_id == cur_msg_id && cur_msg_retired;
+}
+
+// get len of first iov of msg if there is any, otherwise 0
+static size_t msg_capacity(ipc_msg_t *msg) {
```

Comment:
is this a restriction? what are other choices. comment and we should probably try some